### DISCUSSION TITLE
Can this nextcloud plugin be used to "rename" a user?

### DISCUSSION THREAD

**Discussion by narFnarF** [nextcloud/user_migration#249] [Q&A]
Let's say I have an instance with userA and I want to rename it to userB. Currently, Nextcloud don't allow renaming. But maybe this plugin could be used as a workaround?

- UserA exports all data to a file
- Create userB
- UserA shares the exported file to userB (to avoid a long download and a long upload!)
- userB imports the data

Would that work? Any limitation I should know befor attempting? Is there anything that would _not_ be transferred using this method?

**Comment by come-nc** [comment]
> Would that work? Any limitation I should know befor attempting? Is there anything that would _not_ be transferred using this method?

It should work. Make sure you double-check before deleting userA though.

There are thing that will not be transfered using this method, but I cannot list them, all I can do is list what **will** be transferred:
- Files, trashbin, comments, versions
- Calendars
- Contacts
- Settings
- Profile data

Everything else is lost.
Especially:
- Group membership is not migrated
- Shares are not migrated
- Any data from apps with no user_migration support is lost (I only know of forms working on user_migration support, I do not remember if it’s merged in already or not)
- Passwords and tokens are not migrated

Regarding files, you will have trouble with shares and external mounts, as currently they are exported as if normal files and it will attempt to import them, which will duplicate them effectively. This will be fixed by #241 that will ignore shares and external mounts for files export.

So my advice is to actually uncheck files for the export, and use the occ command transfer-ownership for files instead. But I do not know if transfer-ownership takes care of things like trashbin, comments and versions.

Please post here your findings if you test several solutions, that may be helpful for others!

**Reply by narFnarF** [reply]
I tried and as you mentioned, a lot of stuff is not migrated, so to me that's not an acceptable workaround. Maybe in the future if you find a way to really migrate everything.

Thanks for your answer though!
